### PR TITLE
Container guides update

### PR DESCRIPTION
**Pull request by aowen-uwmad**
Quite a few additions to the container documentation. Links below are to the preview pages.

The previous Apptainer guide ([Use Apptainer (Singularity) Environments](https://chtc.github.io/web-preview/preview-aowen-htc-container-update/uw-research-computing/singularity-htc)) is a redirect to the new and updated guides:

* [Use Apptainer Containers](https://chtc.github.io/web-preview/preview-aowen-htc-container-update/uw-research-computing/apptainer-htc)
* [Building an Apptainer Container](https://chtc.github.io/web-preview/preview-aowen-htc-container-update/uw-research-computing/apptainer-build)
* [Advanced Apptainer Example - SUMO](https://chtc.github.io/web-preview/preview-aowen-htc-container-update/uw-research-computing/apptainer-htc-advanced-example)

The "Building an Apptainer Container" guide is linked in both the HTC and HPC sections, as its contents are general to both systems.

Furthermore, created [Using Apptainer Containers on HPC](https://chtc.github.io/web-preview/preview-aowen-htc-container-update/uw-research-computing/apptainer-hpc) guide, and added a deprecation notice to the old/draft version that is still findable (singularity-htc).

### CODE REVIEW

File: _uw-research-computing/apptainer-htc.md [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+[Move the Container .sif File to Staging](#move-the-container-sif-file-to-staging)
+
+[Use an Apptainer Container in HTC Jobs](#use-an-apptainer-container-in-htc-jobs)
+
+{% endcapture %}
+{% include /components/directory.html title="Table of Contents" %}
+
+## Quickstart <a name="quickstart"></a>
+
+### Use an existing container
```
Comment (ChristinaLK) [inline _uw-research-computing/apptainer-htc.md:45]: on further reflection...I might move this to the end and in the "build a container" section, mention that if you already HAVE a container, to skip down. Maybe? 

Let's discuss.

File: _uw-research-computing/apptainer-htc.md [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+[Jump back to Quickstart](#quickstart-build-definition-file)
+
+## Start an Interactive Build Job
+
+Building a container can be a computationally intense process. 
+As such, we require that you only build containers while in an interactive build job.
+On the High Throughput system, you can use the following submit file `build.sub`:
+
+```
```
Comment (ChristinaLK) [inline _uw-research-computing/apptainer-htc.md:196]: Maybe move this up to the shorter section?

**Review by ChristinaLK — APPROVED** [review]
Looks great!!!

File: _uw-research-computing/apptainer-build.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+
+{% endcapture %}
+{% include /components/directory.html title="Table of Contents" %}
+
+## The Apptainer Definition File
+
+The instructions for how Apptainer should build a container are located in the definition file (typically suffixed with `.def`).
+The following table summarizes the sections that are routinely used when constructing a container: 
+
+| Section | Description |
```
[the remark this answers is not in the document]
Comment (aowen-uwmad) [reply _uw-research-computing/apptainer-build.md:40]: Maybe a "Utility" or "Use" column?

```
| `%files` | Specifies files to copy into the container at build time from the host. | Add pre-downloaded source code to the container |
```

File: _uw-research-computing/apptainer-build.md [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+
+Alternatively, 
+
+```
+Bootstrap: docker
+From: nvcr.io/nvidia/tensorflow:24.02-tf2-py3
+```
+
+would use the [NVIDIA TensorFlow 2 v24.02](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/tensorflow/tags) container hosted on the [NVIDIA Container Registry](https://catalog.ngc.nvidia.com/containers) (`nvcr`). 
+
```
Comment (aowen-uwmad) [inline _uw-research-computing/apptainer-build.md:89]:
```suggestion

> If you just want to convert an existing Docker container into an Apptainer container, you do not need to use a definition file.
> Instead, you can directly run the `apptainer build` command using the Docker address, as described [below](#converting-a-docker-image-to-an-apptainer-container-image).

```

File: _uw-research-computing/apptainer-hpc.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+```
+export PATH=/opt/my-program/bin:$PATH
+```
+{:.term}
+
+Consult the "Special Considerations" section of our [Building an Apptainer Container](apptainer-build.html#special-considerations-for-building-your-container) guide for additional information on setting up and testing your container.
+
+When you are finished running commands inside the container, run the command `exit` to exit the container. 
+Your prompt should change back to something like `[username@spark-a006 directory]$`.
```
[the remark this answers is not in the document]
Comment (aowen-uwmad) [reply _uw-research-computing/apptainer-hpc.md:127]:
```suggestion
Your prompt should change back to something like `[username@spark-a006 directory]$`.
If you are satisfied with the container that you built, run the `exit` command again to exit the interactive Slurm session.
```

File: _uw-research-computing/apptainer-build.md [hunk 6]
```diff
@@ -0,0 +1,353 @@
...
+layout: guide
+title: Building an Apptainer Container
+guide:
+    order: 11
+    category: Software Solutions
+    tag:
+        - htc
+        - hpc
+--- 
+
```
[the remark this answers is not in the document]
Comment (aowen-uwmad) [reply _uw-research-computing/apptainer-build.md:12]: Proposed change in https://github.com/CHTC/chtc-website-source/pull/731#discussion_r1543261670

File: _uw-research-computing/apptainer-build.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (aowen-uwmad) [reply _uw-research-computing/apptainer-build.md:47]: Will update description in table with more active language.

File: _uw-research-computing/apptainer-htc.md [hunk 7]
```diff
@@ -0,0 +1,409 @@
...
+<!-- 
+Might be worth setting up the quick-start items as dropdown items.
+Need help from Cannon; default approach does not render well on the website.
+-->
+
+#### 1. Create a definition file <a name="quickstart-build-definition-file"></a>
+
+The definition (`.def`) file contains the instructions for what software to install while building the container.
+CHTC provides example definition files in the `software` folder of our [Recipes GitHub repository](https://github.com/CHTC/recipes). Choose from one of the existing examples, or create your own using the instructions later in this guide.
+
```
[the remark this answers is not in the document]
Comment (aowen-uwmad) [reply _uw-research-computing/apptainer-htc.md:90]: Will add it to the "more information" section later on.

**Pull request merged**